simpleviewinc/tap-events-force
# Zen 363 loading btn

> Note: depends on https://github.com/simpleviewinc/keg-hub/pull/37

**Ticket**: [issue_id](https://jira.simpleviewtools.com/browse/ZEN-363)

## Context
> Note: see this video for reference of expected behavior: https://simpleview-workspace.slack.com/archives/C0149C412E7/p1598981196012400?thread_ts=1598907129.009900&cid=C0149C412E7

* we need to be able to indicate to the user that we're loading/processing something after a button action
* evf guys mentioned that their app uses font awesome spinner. (FontAwesome 4 spinner to be exact)
* The `Processing` button should have:
   * loading indicator on the left
   * text "Processing" on the right

## Goal

* Update the `EvfButton` for a new prop `isProcessing`
* when `isProcessing` is true, change the content to a custom Processing component 

## Updates


* `src/assets/icons/evf/elements/loading.js`
   * added fontAwesome 4 spinner as SVG Component

* `src/components/button/evfButton.js`
   * Updated for new prop `isProcessing`
   * added `Processing` component which contains the spinner and text

* `src/components/loading/evfLoading.js`
  * added Custom Loading component
  * updated buildStyles so it adjusts the width properly when swapping content to `processing`

* `src/containers/testContainer.js`
  * added examples of the evfButtons

* `src/theme/components/modals/groupBookingModal.js`
   * styling update on the bottom section
## Testing

* Build: `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-363-loading-btn command=test:build`
* Tap: `keg evf pack run package=docker.pkg.github.com/simpleviewinc/keg-packages/tap:zen-363-loading-btn`
* navigate to http://local.kegdev.xyz/test
* See the new demo section for EvfButtons
* Click on the `Fetch demo` btn to simulate the interaction
